### PR TITLE
Add `chainerx.hinge`

### PR DESCRIPTION
**Pull request by asi1024**
Takeover #7616.

### CODE REVIEW

File: chainerx_cc/chainerx/routines/loss.cc [hunk 1]
```diff
@@ -31,4 +33,22 @@ Array SigmoidCrossEntropy(const Array& x1, const Array& x2) {
     return -(ignore_mask * (x1 * (x2 - (GreaterEqual(x1, ZerosLike(x1, x1.device()))).AsType(x1.dtype())) - Log1p(Exp(-Absolute(x1)))));
 }
 
+Array Hinge(const Array& x, const Array& t, float norm) {
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/loss.cc]: `float` -> `double`?

Comment (asi1024) [reply chainerx_cc/chainerx/routines/loss.cc]: Fixed.

File: chainerx_cc/chainerx/routines/loss.cc [hunk 2]
```diff
@@ -31,4 +33,22 @@ Array SigmoidCrossEntropy(const Array& x1, const Array& x2) {
     return -(ignore_mask * (x1 * (x2 - (GreaterEqual(x1, ZerosLike(x1, x1.device()))).AsType(x1.dtype())) - Log1p(Exp(-Absolute(x1)))));
 }
 
+Array Hinge(const Array& x, const Array& t, double norm) {
+    if (x.ndim() != 2) {
+        throw DimensionError{"Outputs must be 2 dimensional."};
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/loss.cc]: Outputs -> Input?

File: chainerx_cc/chainerx/routines/loss.h [hunk 3]
```diff
@@ -15,4 +15,7 @@ Array HuberLoss(const Array& x1, const Array& x2, Scalar delta);
 
 Array SigmoidCrossEntropy(const Array& x1, const Array& x2);
 
+// TODO(imanishi): Support `reduce` keyword argument.
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/loss.h]: I guess it won't support the argument.

**Comment by niboshi** [comment]
LGTM
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**